internal/hcs/process.go

```diff
@@ -163,7 +163,42 @@ func (process *Process) Kill(ctx context.Context) (bool, error) {
 		return true, nil
 	}
 
-	resultJSON, err := vmcompute.HcsTerminateProcess(ctx, process.handle)
+	// HCS serializes the signals sent to a target pid per compute system handle.
+	// To avoid SIGKILL being serialized behind other signals, we open a new compute
+	// system handle to deliver the kill signal.
+	// If the calls to opening a new compute system handle fail, we forcefully
+	// terminate the container itself so that no container is left behind
+	hcsSystem, err := OpenComputeSystem(ctx, process.system.id)
+	if err != nil {
+		// log error and force termination of container
+		log.G(ctx).WithField("err", err).Error("OpenComputeSystem() call failed")
+		err = process.system.Terminate(ctx)
+		// if the Terminate() call itself ever failed, log and return error
+		if err != nil {
+			log.G(ctx).WithField("err", err).Error("Terminate() call failed")
+			return false, err
+		}
+		process.system.Close()
+
+		// Return success only if error returned from OpenComputeSystem() call
+		// indicates that the process has already exited. Return false otherwise.
+		if IsNotExist(err) {
+			return true, nil
```

Comment:
This block looks like it will never be reached. If `IsNotExist(err)` is true, then we would have already returned above at the `err != nil` check.

Reply:
Doesn't seem like this is resolved. Did you miss pushing a change?

Reply:
Oops yes I am sorry. I do have the changes locally it wasn't pushed. Doing it in a second

Reply:
Done

Reply:
I think there is a mistake here - At this point, if force terminate of the container goes through in the lines above, then we should return true, nil and not execute the lines after L188